src/Controllers/FeedController.cs

```diff
@@ -32,58 +33,66 @@ public ActionResult RssFeed(string url, int page)
                 var sorterer = new FeedSorterer();
                 sorterer.SortFeed(selectedSort, ref feedItems);
             }
-            int pageSize = int.Parse(_config["FeedPageInfo:pageSize"]);
-            IEnumerable<SyndicationItem> itemsPerPages = feedItems.Skip((page - 1) * pageSize).Take(pageSize);
-            PageInfo pageInfo = new PageInfo { PageNumber = page, PageSize = pageSize, TotalItems = feedItems.Count };
-            Feed rssFeed = _context.Feeds.FirstOrDefault(feed => feed.Url == url);
+            var rssFeed = FormFeed(url, page, feedItems);
+            return View(rssFeed);
+        }
+
+        public Feed FormFeed(string url, int page, List<SyndicationItem> items)
+        {
+            int pageSize = int.Parse(_config.GetSection("FeedPageInfo:pageSize").Value);
+            IEnumerable<SyndicationItem> itemsPerPages = items.Skip((page - 1) * pageSize).Take(pageSize);
+            PageInfo pageInfo = new PageInfo { PageNumber = page, PageSize = pageSize, TotalItems = items.Count };
+            Feed rssFeed = _repository.GetFeedByUrl(url);
             rssFeed.PageInfo = pageInfo;
             rssFeed.SyndicationItems = itemsPerPages;
-            return View(rssFeed);
+            return rssFeed;
         }
 
-        private List<SyndicationItem> GetSyndicationItems(string url)
+        public List<SyndicationItem> GetSyndicationItems(string url)
         {
             XmlReader reader = XmlReader.Create(url);
             SyndicationFeed feed = SyndicationFeed.Load(reader);
             reader.Close();
             return feed.Items.ToList();
         }
 
-        private bool IsSubscribed(string url)
+        public bool IsSubscribed(string url)
         {
-            var contains = _context.Feeds.Include(c => c.Users).FirstOrDefault(feed => feed.Url == url)
-                .Users.Select(u => u.Email).Contains(HttpContext.User.Identity.Name);
-            return contains;
+            var feed = _repository.GetFeedByUrl(url);
```

Comment:
When I reading _repository I can't understand which repository you use name it kinda "_feedRepository" "_userRepository"